Documentation/contributing/development/hive.rst

```diff
@@ -877,4 +877,150 @@ See also the runnable example in `pkg/k8s/resource/example <https://github.com/c
+        )
+
+        lifecycle.Append(ex.jobGroup)
+
+        return &ex
+    }
+
+    func (ex *exampleCell) sync(ctx context.Context) error {
+        for i := 0; i < 3; i++ {
+            if err := ex.doSomeWork(); err != nil {
+                return fmt.Errorf("doSomeWork: %w", err)
+            }
+        }
+
+        return nil
+    }
+
+    func (ex *exampleCell) daemon(ctx context.Context) error {
+        for {
+            randomTimeout := time.NewTimer(time.Duration(rand.Intn(3000)) * time.Millisecond)
+            select {
+            case <-ctx.Done():
+                return nil
+
+            case <-randomTimeout.C:
+                ex.doSomeWork()
+            }
+        }
+    }
+
+    func (ex *exampleCell) timer(ctx context.Context) error {
+        if err := ex.doSomeWork(); err != nil {
+            return fmt.Errorf("doSomeWork: %w", err)
+        }
+
+        return nil
+    }
+
+    func (ex *exampleCell) Trigger() {
+        ex.trigger.Trigger()
+    }
+
+    func (ex *exampleCell) observer(ctx context.Context, event struct{}) error {
+        ex.logger.Info("Observed event")
+        return nil
+    }
+
+    func (ex *exampleCell) HeavyLifting() {
+        ex.jobGroup.Add(job.OneShot("long-running-job", func(ctx context.Context) error {
+            for i := 0; i < 1_000_000; i++ {
+                // Do some heavy lifting
+            }
+            return nil
+        }))
+    }
+
+    func (ex *exampleCell) doSomeWork() error {
+        ex.workChan <- struct{}{}
+        return nil
+    }
+
 
+The above example shows a number of use-cases in one cell. We start by requesting the job.Registry
+via the constructor. We can use the registry to create job groups, in most cases one will be enough.
+To this group we can add our jobs in the constructor. Any jobs added in the constructor are queued
+until the lifecycle of our cell starts. The group is added to the lifecycle and manages this 
+internally. Jobs can also be added at runtime which can be handy for dynamic workloads while still
+guaranteeing a clean shutdown.
+
+A job group will cancel the context to all jobs when the lifecycle ends. Any job callbacks are 
+expected to exit as soon as possible when the ``ctx`` is "Done". The group will make sure that all
+jobs are properly shutdown before the cell stops. If callbacks that do not stop within reasonable 
+amount of time may cause hive to perform a hard shutdown.
+
+There are 3 job types: one-shot jobs, timer jobs, and observer jobs. One shot jobs run a limited 
+amount of times, they can be used for short running jobs or jobs that span the entire lifecycle.
+Once the callback exits without error, its never called again. A one-shot can optionally have retry
+logic and/or trigger hive shutdown if it fails. Timers are called on a specified interval but they
+can also be externally triggered. Lastly, we have observer jobs which are invoked for every event
+on a ``stream.Observable``.
```

Comment:
Consistent hyphens, grammar
```suggestion
There are 3 job types: one-shot jobs, timer jobs, and observer jobs. One-shot jobs run a limited 
number of times: you can use them for brief jobs, or for jobs that span the entire lifecycle.
Once the callback exits without error, it is never called again. Optionally, a one-shot job can include retry
logic and/or trigger hive shutdown if it fails. Timers are called on a specified interval but they
can also be externally triggered. Lastly, observer jobs are invoked for every event
on a ``stream.Observable``.
```

Reply:
@zacharysarah thoughts on "use them" vs. "you can use them"? The former seems more concise and imperative.

(Otherwise all of this feedback looks good, @dylandreimerink / @zacharysarah do you think either of you would be able to apply these diffs in a fresh PR to the tree?)

Reply:
@joestringer Sorry, I'm just now seeing this comment. You could use either, but "use them" is totally appropriate here, and I kinda like it.